### ISSUE TITLE
[CLI]: channel config populated with invalid version when using version channel

### ISSUE THREAD

**Issue by LanceEa** [infinyon/fluvio#2168]
**What happened**
When using the command `fluvio version create 123` an error will be thrown because the version cannot be resolved.  Despite not being a valid version the `channel` config file is being updated with the channel key as seen here:

```toml
current_channel = "stable"
[channel.123]
binary_location = "/Users/lance/.fluvio/bin/fluvio-123"
extensions = "/Users/lance/.fluvio/bin/extensions-123"
image_tag_strategy = "Version"

[channel.stable]
binary_location = "/Users/lance/.fluvio/bin/fluvio-stable"
extensions = "/Users/lance/.fluvio/extensions"
image_tag_strategy = "Version"

[channel."0.9.18"]
binary_location = "/Users/lance/.fluvio/bin/fluvio-0.9.18"
extensions = "/Users/lance/.fluvio/bin/extensions-0.9.18"
```

Also, running `fluvio version list` shows the invalid version:

```bash
fluvio version list
  0.9.18
  123
* stable
```

**Expected behavior**

I would expect the resolution of the version to happen prior to making any changes to the channel config so as not to pollute the available versions.

**Describe the setup**
- local fluvio cli installed
- `fluvio version`
Release Channel      : stable
Fluvio CLI           : 0.9.18
Fluvio CLI SHA256    : f97262eec5fcb1f0b05c3cd6d1a7cdf40cfe83d1c5b702be49b82aa4575f409e
Fluvio Platform      : 0.9.18 (kind-fluvio)
Git Commit           : 2b6397f56e3935807ebb9e84c60517f815426d51
OS Details           : Darwin 12.1 (kernel 21.2.0)
=== Plugin Versions ===
Fluvio Runner (fluvio-run)     : 0.0.0
Infinyon Cloud CLI (fluvio-cloud) : 0.1.6

**How to reproduce it (as minimally and precisely as possible)**
Steps to reproduce the behavior on Mac:
1. Run the command `fluvio version create 1234`
2. An unresolved version error message will be shown
3. cat ~/.fluvio/channel
4. run the command `fluvio version list`

You will notice the bad version has been written to the config file

**Environment (please complete the following information):**
 - OS: Mac M1
 - Fluvio Version:  0.9.18

**Comment by rrrodzilla** [comment]
This appears to be a [one-line fix](https://github.com/infinyon/fluvio/blob/e3f20038fbc54c1e936502174f917cd616dd7184/crates/fluvio-channel-cli/src/cli/create.rs#L108) (move line 108 to line 115).  I'll write a test to regress the fix and submit the PR this evening.

**Comment by tjtelan** [comment]
Closed by #2170
